src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2184,7 +2186,7 @@ private void WriteStringToAllListeners(string eventName, string msg)
                 EventName = eventName,
                 Message = msg,
                 Payload = new ReadOnlyCollection<object?>(new object[] { msg }),
-                PayloadNames = new ReadOnlyCollection<string>(new string[] { "message" })
+                PayloadNames = new ReadOnlyCollection<string>(s_payloadNames)
```

Comment:
It'd be nice if this one could be lazily-allocated, e.g.
```C#
private static string[]? s_errorPayloadNames;
...
PayloadNames = s_errorPayloadNames ??= new ReadOnlyCollection<string>(new string[] { "message" }),
```
however the analyzer doesn't currently recognize that this array is being used to initialize a static and thus still complains.  Can we fix the analyzer?

Regardless, we should cache the ReadOnlyCollection rather than caching just the array.

Reply:
> Regardless, we should cache the ReadOnlyCollection rather than caching just the array.

Done

> however the analyzer doesn't currently recognize that this array is being used to initialize a static and thus still complains. Can we fix the analyzer?

We have excluded `static readonly` fields/properties, we could also add `static ReadOnlyCollection<T>` to the exclusion filed https://github.com/dotnet/roslyn-analyzers/issues/6629

Reply:
Would it make sense instead to just exclude assignments to all static fields/properties?